src/atomic.rs

```diff
@@ -328,10 +335,22 @@ impl AtomicBlock {
 
 impl Debug for AtomicBlock {
     fn fmt(&self, f: &mut Formatter) -> Result<(), FormatError> {
-        f.debug_struct("AtomicBlock")
-            .field("mask", &self.mask)
-            .field("atom", &self.atom.get().unwrap().iter())
-            .finish()
+        let mut s = f.debug_struct("AtomicBlock");
+        s.field("mask", &self.mask);
+        if let Some(atom) = self.atom.get() {
+            s.field(
+                "atom",
+                &atom
+                    .iter()
+                    .enumerate()
+                    .filter_map(|(idx, v)| match v.fetch_or(0, Ordering::Relaxed) {
+                        0 => None,
+                        x => Some((idx, x)),
+                    })
+                    .collect::<Vec<(usize, usize)>>(),
+            );
+        }
```

Comment:
A helper type could be used here to avoid the allocation when collecting into a vec:

```suggestion
        struct DebugAtom<'a>(&'a [AtomicUsize; 1 << BITS]);
        impl Debug for DebugAtom<'_> {
            fn fmt(&self, f: &mut Formatter) -> Result<(), FormatError> {
                f.debug_list()
                    .entries(self.0.iter().enumerate().filter_map(|(idx, v)| {
                        match v.load(Ordering::Relaxed) {
                            0 => None,
                            x => Some((idx, x)),
                        }
                    }))
                    .finish()
            }
        }

        let mut s = f.debug_struct("AtomicBlock");
        s.field("mask", &self.mask);
        if let Some(atom) = self.atom.get() {
            s.field("atom", &DebugAtom(atom));
        }
```